### DISCUSSION TITLE
Rule and DataFrame datatypes not matching

### DISCUSSION THREAD

**Discussion by FShinoda** [canimus/cuallee#105] [Q&A] [answered]
Hey there!

I am getting the following error:
<img width="562" alt="image" src="https://github.com/canimus/cuallee/assets/45886482/1b3f6cfb-3873-48b5-9c51-f5c6550cd95b">

and yeah, `acc_benefits_id` is not string. it is long datatype. 

How can I define the rules so it considers this column like long or some related datatype? 

Thanks in advance!!

**Comment by canimus** [comment, marked as answer]
Hello, happy to help. What kind of `check` are you performing? I assume that is `has_pattern` as it is the only one with a string requirement.
And yes, certain checks are associated to specific data types as shown in the README. One alternative is to simply cast your column:
```python
# assuming your dataframe as `acc_benefits_id`
from cuallee import Check, CheckLevel
check = Check(CheckLevel.WARNING, "Pattern")
check.has_pattern("acc_benefits_id_str", "your_pattern")
check.validate(df.select("acc_benefits_id").cast("string").alias("acc_benefits_id_str")).show()
```

However, I do think that a pattern check in a numeric data type, is probably not what you are searching for isn't?
Is computationally more expensive, or probably I am just wildly making assumptions?
Ok. Hope it helps @FShinoda 
Regards,
